Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py

```diff
@@ -0,0 +1,614 @@
+        return DBotScoreReliability.D
+    if confidence == 0:
+        return DBotScoreReliability.E
+    else:
+        return DBotScoreReliability.F
+
+
+def get_tlp(object_marking_refs: list[str], stix_bundle: dict) -> str:
+    """
+    Retrieve marking-definition object from object_marking_refs and return TLP
+    If no TLP where found, return "RED" by default
+    """
+    for object_marking_ref in object_marking_refs:
+        for stix_object in stix_bundle["objects"]:
+            if stix_object["id"] == object_marking_ref:
+                return stix_object["definition"]["tlp"]
+
+    return "red"
+
+
+def get_stix_object_reputation(stix_bundle: dict, stix_object: dict) -> Optional[CommandResults]:
+    """ "
+    Transform a STIX object into a Cortex XSOAR indicator
+    """
+
+    reputation_score: int = get_reputation_score(stix_object.get("indicator_types", []))
+    reliability_score: str = get_reliability_score(int(stix_object.get("confidence", -1)))
+    tlp: str = get_tlp(stix_object["object_marking_refs"], stix_bundle)
+
+    if "ipv4-addr" in stix_object["x_ic_observable_types"] or "ipv6-addr" in stix_object["x_ic_observable_types"]:
+        return get_ip_indicator_reputation(stix_object, reputation_score, reliability_score, tlp)
+    if "file" in stix_object["x_ic_observable_types"]:
+        return get_file_indicator_reputation(stix_object, reputation_score, reliability_score, tlp)
+    if "domain-name" in stix_object["x_ic_observable_types"]:
+        return get_domain_indicator_reputation(stix_object, reputation_score)
+    if "url" in stix_object["x_ic_observable_types"]:
+        return get_url_indicator_reputation(stix_object, reputation_score)
+
+    return None
+
+
+def get_ip_indicator_reputation(stix_object: dict, reputation_score: int, reliability_score: str, tlp: str) -> CommandResults:
+    """
+    Return stix_object of type IP as indicator
+    """
+    dbot_score = Common.DBotScore(
+        indicator=stix_object["name"],
+        indicator_type=DBotScoreType.IP,
+        integration_name=INTEGRATION_NAME,
+        score=reputation_score,
+        reliability=reliability_score,
+    )
+
+    ip = Common.IP(
+        ip=stix_object["name"],
+        dbot_score=dbot_score,
+        traffic_light_protocol=tlp,
+    )
+    return CommandResults(
+        outputs_prefix="SEKOIAIntelligenceCenter.IP",
+        outputs_key_field="name",
+        outputs=stix_object,
+        indicator=ip,
+    )
+
+
+def get_file_indicator_reputation(stix_object: dict, reputation_score: int, reliability_score: str, tlp: str) -> CommandResults:
+    """
+    Return stix_object of type file as indicator
+    """
+
+    dbot_score = Common.DBotScore(
+        indicator=stix_object["name"],
+        indicator_type=DBotScoreType.FILE,
+        integration_name=INTEGRATION_NAME,
+        score=reputation_score,
+        reliability=reliability_score,
+    )
+
+    hashes = extract_file_indicator_hashes(stix_object["name"])
```

Comment:
I think it's preferable to use `pattern` instead of `name` (that could be other than a stix-patterning)
```suggestion
    hashes = extract_file_indicator_hashes(stix_object["pattern"])
```

Reply:
done